### PR TITLE
chore: new spfx scaffolding question for package selection

### PR DESCRIPTION
**Pull request by yuqizhou77**
- add new questions for SPFx package selection (temporarily put it behind a feature flag)
image

- If user currently does not have any SPFx installed globally, we won't show version details
image

- Since getting versions of latest or installed SPFx will take some time, we show a progress indicator when loading (similar to the UI we have today when checking NPM and Node version)
image

### CODE REVIEW

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -187,6 +187,12 @@
   "plugins.spfx.questions.versionCheck.title": "Checking Node.js and NPM versions...",
   "plugins.spfx.questions.webpartName.error.duplicate": "The folder %s already exists. Choose a different name for your component.",
   "plugins.spfx.questions.webpartName.error.notMatch": "%s doesn't match pattern: %s",
+  "plugins.spfx.questions.packageSelect.title": "SharePoint Framework",
+  "plugins.spfx.questions.packageSelect.placeholder": "Select an option for scaffolding",
+  "plugins.spfx.questions.packageSelect.useGlobalPackage.withVersion.label": "Use globally installed SPFx (%s)",
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/resource/package.nls.json:192]: Add 'v' prefix? Need PM confirm here.

File: packages/fx-core/src/component/resource/spfx/utils/constants.ts [hunk 2]
```diff
@@ -26,6 +26,9 @@ export class Constants {
   public static readonly SUPPORTED_NPM_VERSION = ["7", "8"];
   public static readonly SPFX_HELP_LINK = "https://aka.ms/teamsfx-spfx-help";
   public static readonly TEMPLATE_NAME = "spfx-tab";
+  public static readonly LatestVersion = "latest";
+  public static readonly RecommenLowestSpfxVersion = "v1.14.0";
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/src/component/resource/spfx/utils/constants.ts:30]: Typo: Recommend

**Comment by HuihuiWu-Microsoft** [comment]
Since user is able to go back to last step in question flow, is everything OK in that case?

**Comment by yuqizhou77** [comment]
> Since user is able to go back to last step in question flow, is everything OK in that case?

Yes, similar to checking Node and Npm Version, user won't go back to the question for us to load versions.

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:192]: updated. thought it would be consistent with the description message we put.

Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:192]: updated the code again. and will add "v" in code when generating string. Please see the attached screenshot in PR description.

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -187,6 +187,13 @@
...
+  "plugins.spfx.questions.packageSelect.useGlobalPackage.noVersion.label": "Use globally installed SPFx",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:193]: Please check whether this "SPFx" is publicly communicated to end users/developers. If not through their official documents or sources, we should always avoid using acronyms, but to use the full name instead whenever possible.

Comment (sffamily) [reply packages/fx-core/resource/package.nls.json:193]: The finding will apply to all occurences of SPFx.

File: packages/fx-core/resource/package.nls.json [hunk 4]
```diff
@@ -187,6 +187,13 @@
...
+  "plugins.spfx.questions.packageSelect.useGlobalPackage.detail": "Recommend SPFx %s or later",
+  "plugins.spfx.questions.packageSelect.installLocally.withVersion.label": "Install latest SPFx (%s) locally in Teams Toolkit directory ",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:195]: latest -> the latest

File: packages/fx-core/resource/package.nls.json [hunk 5]
```diff
@@ -187,6 +187,13 @@
...
+  "plugins.spfx.questions.packageSelect.installLocally.noVersion.label": "Install latest SPFx locally in Teams Toolkit directory ",
```
Comment (sffamily) [inline packages/fx-core/resource/package.nls.json:196]: latest -> the latest

File: packages/fx-core/resource/package.nls.json (hunk 3 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json]: Updated to "SharePoint Framework" as what spfx doc uses.

**Review by sffamily — APPROVED** [review]

File: packages/fx-core/resource/package.nls.json (hunk 3 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json]: synced offline and we will confirm with SPFx team about whether "SPFx" is enough.

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Pull request merged**